bionic/core/flow_execution.py

```diff
@@ -407,47 +410,54 @@ def _mark_entry_active(self, entry):
 
     def _wait_on_in_progress_entries(self):
         "Waits on any in-progress entry to finish."
-        futures = set(entry.future for entry in self._in_progress_entries)
-        finished_futures, _ = wait(futures, return_when=FIRST_COMPLETED)
-        try:
-            for finished_future in finished_futures:
+        future_to_entry = {entry.future: entry for entry in self._in_progress_entries}
```

Comment:
Nit: the convention for dicts here would be `entries_by_future`.

Reply:
Good point. Renamed variable.